### PR TITLE
Automated adoption of Octavia certificates

### PR DESCRIPTION
**Pull request by weinimo**
Depends-On: https://review.rdoproject.org/r/c/rdo-jobs/+/54139
Jira: OSPNET-1069

### CODE REVIEW

**Comment by weinimo** [comment]
recheck

**Comment by weinimo** [comment]
recheck

**Comment by weinimo** [comment]
recheck

File: docs_user/modules/proc_adopting-the-loadbalancer-service.adoc
Comment (gthiemonge) [inline docs_user/modules/proc_adopting-the-loadbalancer-service.adoc:9]: maybe you should put an example of the format of those variables (like `ssh -i /path/to/key root@controller-1.xxx` or any useful tip)

File: tests/roles/dataplane_adoption/tasks/octavia_certs.yaml
Comment (gthiemonge) [inline tests/roles/dataplane_adoption/tasks/octavia_certs.yaml:7]: I'm wondering if the file from **config-data/puppet-generated** would be more appropriate here

File: tests/roles/dataplane_adoption/tasks/octavia_certs.yaml
Comment (gthiemonge) [inline tests/roles/dataplane_adoption/tasks/octavia_certs.yaml:8]: why is xargs needed?

Comment (weinimo) [reply tests/roles/dataplane_adoption/tasks/octavia_certs.yaml:8]: to remove whitespace

File: docs_user/modules/proc_adopting-the-loadbalancer-service.adoc
Comment (weinimo) [reply docs_user/modules/proc_adopting-the-loadbalancer-service.adoc:9]: I tried to explain it a bit better now. You'll see it in my next commit. I don't want to push it now because I don't want the CI jobs to restart before they finish.

File: tests/roles/dataplane_adoption/tasks/octavia_certs.yaml
Comment (weinimo) [reply tests/roles/dataplane_adoption/tasks/octavia_certs.yaml:7]: I changed the path now. It is the same file, and I noticed that some other certificate related files were also stored in different locations and I wasn't always sure which location is best. Again, I'll push the change soon.

File: tests/roles/dataplane_adoption/tasks/octavia_certs.yaml
Comment (weinimo) [inline tests/roles/dataplane_adoption/tasks/octavia_certs.yaml:22]: Does `/tmp/octavia-ssl/` still exist after controller reboot?

File: tests/roles/dataplane_adoption/tasks/octavia_certs.yaml
Comment (weinimo) [inline tests/roles/dataplane_adoption/tasks/octavia_certs.yaml:46]: TODO remove that line

File: docs_user/modules/proc_adopting-the-loadbalancer-service.adoc
Comment (weinimo) [reply docs_user/modules/proc_adopting-the-loadbalancer-service.adoc:9]: I pushed the change in the meantime.

**Comment by cescgina** [comment]
Since this is now passing CI, just wanted to note that we have to merge this in unison with the depends-on patch https://review.rdoproject.org/r/c/rdo-jobs/+/54139, if we merge only one of them, we'll break the existing jobs

File: tests/roles/common_defaults/defaults/main.yaml
Comment (marios) [inline tests/roles/common_defaults/defaults/main.yaml:16]: sanity check. do we really mean to default to on here ? 
on the other side in rdo-jobs you default false
```
            -e "enable_octavia={{ enable_octavia | default('false') }}"
```

**Review by marios — COMMENTED** [review]
waiting to hear about testing (especially the multinodes) is there a testproject?

we need to coordinate with the depends-on

File: tests/roles/common_defaults/defaults/main.yaml
Comment (weinimo) [reply tests/roles/common_defaults/defaults/main.yaml:16]: It has been disabled because of a CIX: https://issues.redhat.com/browse/OSPCIX-420 . It crashed tempest. I think we should keep it enabled per default if possible in order to test it upsteam at least.
@cescgina

File: tests/roles/dataplane_adoption/tasks/octavia_certs.yaml
Comment (weinimo) [reply tests/roles/dataplane_adoption/tasks/octavia_certs.yaml:22]: I tried it out. At least in the standlone deployment the path and the files in it still exist after reboot

File: tests/roles/common_defaults/defaults/main.yaml
Comment (gthiemonge) [reply tests/roles/common_defaults/defaults/main.yaml:16]: It crashed tempest because Octavia was disabled, enabling it during the deployment and the adoption will fix it

File: tests/roles/dataplane_adoption/tasks/main.yaml
Comment (marios) [inline tests/roles/dataplane_adoption/tasks/main.yaml:535]: these are passing in the last zuul run here for the standalone 


        * https://logserver.rdoproject.org/56/556/e29930087fc1e324212e497e62c74e3b12337c1a/github-check/adoption-standalone-to-crc-ceph/4538bfa/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_with_ceph_out_2024-10-08T08:41:48EDT.log
        * TASK [dataplane_adoption : Adopt Octavia Certificates Wallaby->Antelope FFU] ***
included: /home/zuul/src/github.com/openstack-k8s-operators/data-plane-adoption/tests/roles/dataplane_adoption/tasks/octavia_certs.yaml for localhost

**Review by marios — COMMENTED** [review]
before we merge, and if we have not already, can we please sanity check this with the multinodes since the depends-on is enabling for the multinode jobs too

**Comment by weinimo** [comment]
recheck

**Review by marios — APPROVED** [review]
lgtm and the multinode test passed the octavia adoption https://logserver.rdoproject.org/15/53815/4/check/periodic-adoption-multinode-to-crc-no-ceph/65282cf/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2024-10-10T15:01:23EDT.log 

please see the comment on depends-on about the default value though lets be consistent unless there is some reason to not be

**Review by cescgina — APPROVED** [review]
lgtm

**Review by jistr — APPROVED** [review]

**Comment by marios** [comment]
this seems ready to merge - is it really necessary to have four commits here though ? Personally I'd prefer them squashed into one before merging but I am not going to block the merge on that

**Comment by weinimo** [comment]
> this seems ready to merge - is it really necessary to have four commits here though ? Personally I'd prefer them squashed into one before merging but I am not going to block the merge on that

Let's merge it ASAP. I could squash it, but that would further delay the merge and this has been open for too long already IMO.

File: docs_user/modules/proc_adopting-the-loadbalancer-service.adoc
Comment (leifmadsen) [inline docs_user/modules/proc_adopting-the-loadbalancer-service.adoc:30]: I would probably comment this out

File: docs_user/modules/proc_adopting-the-loadbalancer-service.adoc
Comment (leifmadsen) [inline docs_user/modules/proc_adopting-the-loadbalancer-service.adoc:37]: Traditionally in docs, console commands are lead by `$` or `#`.

**Pull request merged**
